packages/flutter_tools/lib/src/ios/ios_deploy.dart

```diff
@@ -169,30 +211,193 @@ class IOSDeploy {
+    final Completer<bool> debuggerCompleter = Completer<bool>();
+    try {
+      _iosDeployProcess = await _processUtils.start(
+        _launchCommand,
+        environment: _iosDeployEnv,
+      );
+      bool lldbRunCalled = false;
+      String lastLineFromDebugger;
+      final StreamSubscription<String> stdoutSubscription = _iosDeployProcess.stdout
+          .transform<String>(utf8.decoder)
+          .transform<String>(const LineSplitter())
+          .listen((String line) {
+        _monitorIOSDeployFailure(line, _logger);
+
+        // (lldb)     run
+        // success
+        // 2020-09-15 13:42:25.185474-0700 Runner[477:181141] flutter: Observatory listening on http://127.0.0.1:57782/
+        if (line == '(lldb)     run') {
+          _logger.printTrace(line);
+          lldbRunCalled = true;
+          return;
+        }
+        // next line must be "success".
+        if (lldbRunCalled && line == 'success') {
+          _logger.printTrace(line);
+          _debuggerAttached = true;
+          debuggerCompleter.complete(true);
+          return;
+        }
+        lldbRunCalled = false;
+        if (line.contains('PROCESS_STOPPED') ||
+            line.contains('PROCESS_EXITED')) {
+          // The app exited or crashed, so stop echoing the output.
+          // Don't pass any further ios-deploy debugging messages to the log reader after it exits.
+          _debuggerAttached = false;
+          _logger.printTrace(line);
+          return;
+        }
+
+        if (line.contains('error: process launch failed')) {
+          // Timed out waiting for the app to launch.
+          _debuggerAttached = false;
+          _logger.printTrace(line);
+          debuggerCompleter.complete(false);
+          return;
+        }
+        if (!debuggerAttached) {
+          _logger.printTrace(line);
+          return;
+        }
+        if (lastLineFromDebugger != null && lastLineFromDebugger.isNotEmpty && line.isEmpty) {
+          // The lldb console stream from ios-deploy is separated lines by an extra \r\n.
+          // To avoid all lines being double spaced, if the last line from the
+          // debugger was not an empty line, skip this empty line.
+          // This will still cause "legit" logged newlines to be doubled...
+        } else {
+          _debuggerOutput.add(line);
+        }
+        lastLineFromDebugger = line;
+      });
+      final StreamSubscription<String> stderrSubscription = _iosDeployProcess.stderr
+          .transform<String>(utf8.decoder)
+          .transform<String>(const LineSplitter())
+          .listen((String line) {
+        _logger.printTrace(line);
+      });
+      unawaited(_iosDeployProcess.exitCode.then((int status) {
+        _logger.printTrace('ios-deploy exited with code $exitCode');
+        _debuggerAttached = false;
+        unawaited(stdoutSubscription.cancel());
+        unawaited(stderrSubscription.cancel());
```

Comment:
Does ios-deploy have anything interesting on its stderr when it exits with an error?

Reply:
Some stuff:
https://github.com/ios-control/ios-deploy/blob/master/src/ios-deploy/ios-deploy.m#L1741
Though nothing that's probably actionable to the developer.  It's being `printTrace`d a few lines above this.
https://github.com/flutter/flutter/pull/66092/files/e91e25e01ffa596f7e60c861b435a9c672a9a29e#diff-8d108d6a887a4f5b74f7c7d77003e795R326

They seem to put actionable errors to `stdout` (see `_monitorIOSDeployFailure` parsing below, which existed before this PR).